includes/admin/class-wcmp-export-consignments.php

```diff
@@ -471,4 +471,35 @@ private function setBaseData(): void
             ->setDeliveryType($this->getDeliveryType())
             ->setLabelDescription($this->getFormattedLabelDescription());
     }
+
+    /**
+     * @throws Exception
+     */
+    public function validate(): bool
+    {
+        if (! $this->consignment->validate()) {
+            return false;
+        }
+
+        $weight         = $this->getTotalWeight(); // NOTE this is already the weight per consignment (in grammes)
+        $maxColloWeight = WCMP_Data::MAX_COLLO_WEIGHT_PER_PACKAGE_TYPE[$this->getPackageType()];
+        if (! $maxColloWeight) {
+            $maxColloWeight = WCMP_Data::MAX_COLLO_WEIGHT_PER_PACKAGE_DEFAULT;
+        }
+
+        if ($weight > $maxColloWeight) {
+            $translationKey = (1 === $this->orderSettings->getColloAmount())
+                ? 'error_collo_weight_%1$s_but_max_%2$s'
+                : 'error_colli_weight_%1$s_but_max_%2$s';
+            $message        = sprintf(
+                __($translationKey, 'woocommerce-myparcel'),
+                $weight / 1000,
+                $maxColloWeight / 1000
+            );
+            $hint           = __('export_hint_change_parcel', 'woocommerce-myparcel');
+            throw new Exception("{$message} {$hint}");
+        }
```

Comment:
Ik denk dat je beter regel 484–501 in een nieuwe method kunt extracten. Deze functie heet immers gewoon `validate` dus het lijkt mij mooi als dit brok verdwijnt naar een `validateWeight` of iets dergelijks.

Reply:
done